src/wallet/test/wallet_tests.cpp

```diff
@@ -605,7 +605,7 @@ BOOST_FIXTURE_TEST_CASE(ListCoinsTest, ListCoinsTestingSetup)
     // returns the coin associated with the change address underneath the
     // coinbaseKey pubkey, even though the change address has a different
     // pubkey.
-    AddTx(CRecipient{GetScriptForRawPubKey({}), 1 * COIN, /*subtract_fee=*/false});
+    AddTx(CRecipient{CPubKey{}, 1 * COIN, /*subtract_fee=*/false});
```

Comment:
Not for this PR but based on this recent not so good experience, I think we should make all destinations constructors explicit.

Here, we are providing a `CPubKey` to implicitly construct a `PubKeyDestination`, which is not very intuitive. There is a `PKHash` constructor (the CTxDestination type of a p2pkh) that also accepts an `CPubKey` as argument.